synapse/storage/databases/main/relations.py

```diff
@@ -132,41 +132,66 @@ def _get_recent_references_for_event_txn(
             "get_recent_references_for_event", _get_recent_references_for_event_txn
         )
 
-    async def event_has_relations(self, event_id: str) -> bool:
-        """Check if a given event has relations.
+    async def event_includes_relation(self, event_id: str) -> bool:
+        """Check if the given event includes a valid relation.
+
+        An event has a relation if it has a valid m.relates_to with a rel_type
+        and event_id in the content:
+
+        {
+            "content": {
+                "m.relates_to": {
+                    "rel_type": "m.replace",
+                    "event_id": "$other_event_id"
+                }
+            }
+        }
```

Comment:
The small piece of mathematician in me thinks the terminology is off (probably not your fault so much as the spec's or even English in general — maybe relation has more meanings that confuse this), but the example paints a thousand words, so thanks :)

Reply:
I was thinking a bit more about this last night and we seem to be missing ones for the target event of the relation vs. the event which includes the relation? Definitely open to ideas if you have any!

Reply:
if I understand you properly:

> to be missing ones [words?] for the target event of the relation vs. the event which includes the relation

parent and child are possibly my favourite for now, and the intuition that the parent existed first matches up with reality.

If this isn't what you were asking about the please rephrase :)

Reply:
Yes, I meant `words` not `ones`. I've edited my comment!